### PR TITLE
[issue-3341][helm] Move tests to CI folder and fix YAML linting

### LINKED ISSUE #3341: [helm] The helm chart has inconsistent indentation
After raising a couple of PRs I've noticed that the Helm chart contains inconsistent indentation.

Would it be ok to raise a PR using [`yamllint`](https://github.com/adrienverge/yamllint), [used by Helm chart testing](https://github.com/helm/chart-testing#prerequisites) to detect this after generating the tests?

Alternatively, the whole project could use [`pre-commit`](https://pre-commit.com)

### PR DESCRIPTION
**Pull request by carlosjgp**
#### PR Description

Move tests to folder `operations/helm/charts/grafana-agent/tests` to `operations/helm/charts/grafana-agent/ci` to take advantage of [chart-testing cli feature](https://github.com/helm/chart-testing/blob/v3.8.0/doc/ct_lint.md)
> Charts may have multiple custom values files matching the glob pattern '*-values.yaml' in a directory named 'ci' in the root of the chart's directory. The chart is linted for each of these files. If no custom values file is present, the chart is linted with defaults.
```
$ cd operations/helm/charts/grafana-agent
$ mv tests ci
$ cd ci
$ for f in *.values.yaml; do mv "$f" "$(echo "$f" | sed s/.values.yaml/-values.yaml/)"; done
$ popd
$ popd
$ make generate-helm-tests
```

I also added `operations/helm/lintconf.yaml` from [helm chart testing binary distribution](https://github.com/helm/chart-testing/releases) for `yamlint`

Changed the `rebuild-tests.sh` to dynamically discover the location of the charts and added `yamlint` after generating the tests

And finally addressed all the lining errors.

Helm ct does perform some YAML linting but only on `Chart.yaml`, `values.yaml` and `ci/*-values.yaml` but we want to lint the output of the chart instead

#### Which issue(s) this PR fixes

Fixes #3341

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: operations/helm/charts/grafana-agent/config/example.river
Comment (carlosjgp) [inline operations/helm/charts/grafana-agent/config/example.river:1]: I didn't manage to be able to trim whitespace of newlines on files and `yamlint` was not happy about it. Removing the new lines makes the linter happy
See https://github.com/helm/helm/issues/11955

File: operations/helm/charts/grafana-agent/templates/configmap.yaml [hunk 1]
```diff
@@ -8,15 +8,15 @@ metadata:
 data:
   {{- if eq .Values.agent.mode "flow" }}
   {{- if .Values.agent.configMap.content }}
-  config.river: {{ toYaml (tpl  .Values.agent.configMap.content .) | indent 2 }}
+  config.river: |- {{- toYaml (tpl  .Values.agent.configMap.content .) | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (carlosjgp) [reply operations/helm/charts/grafana-agent/templates/configmap.yaml:11]: No changes on the user side are required I just realised that we are adding one too many `|-` and that comes from the `toYaml`

I've pushed a the fix for it

[the remark this answers is not in the document]
Comment (rfratto) [reply operations/helm/charts/grafana-agent/templates/configmap.yaml]: Hm, I'm confused; you had to remove the newlines in config/example.river to get it to render at all, right? So wouldn't users have to do the same for their config files?

[the remark this answers is not in the document]
Comment (carlosjgp) [reply operations/helm/charts/grafana-agent/templates/configmap.yaml]: I see. I got confused by the line the comment is on.

Users are not affected by any of the changes on this PR the changes on the `example.river` file is just to pass the linting of the chart. Users can use new lines as they wish

There might be a way to exclude the ConfigMap file to linted if this is a problem or confusing

[the remark this answers is not in the document]
Comment (carlosjgp) [reply operations/helm/charts/grafana-agent/templates/configmap.yaml]: I've pushed the change for the linter to ignore whitespaces on the configmap

I hope the changes are more clear now

File: operations/helm/charts/grafana-agent/CHANGELOG.md [hunk 2]
```diff
@@ -15,10 +15,13 @@ Unreleased
 - Add support for adding Annotations to Service (@ofirshtrull)
 - Add `agent.envFrom` value. (@carlosjgp)
 - Add `controller.hostNetwork` value. (@carlosjgp)
+- Move and rename tests to make use of Helm chart-testing. (@carlosjgp)
+- Lint the YAML output of the tests. (@carlosjgp)
```
Comment (rfratto) [inline operations/helm/charts/grafana-agent/CHANGELOG.md:19]: These two aren't user-facing changes to the chart, so typically we'd leave them out of the chagelog (even though we do appreciate the PR).

```suggestion
```

**Review by rfratto — APPROVED** [review]
LGTM, thank you!

**Pull request merged**